### PR TITLE
feat: add iPXE build with embedded placeholder script

### PR DESCRIPTION
**Pull request by smira**
This builds iPXE with HTTPS enabled, and embedded placeholder script
which is to be replaced with the real script during Sidero
metal-controller-manager bootstrap.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: ipxe/pkg.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        {{ if eq .ARCH "aarch64" }}
+        cp -p bin-arm64-efi/ipxe.efi /rootfs/usr/libexec
+        {{ else }}
+        cp -p bin/undionly.kpxe bin-x86_64-efi/ipxe.efi /rootfs/usr/libexec
+        cp -p bin/undionly.kpxe.bin bin/undionly.kpxe.zinfo util/zbin /rootfs/usr/libexec/zbin
+        {{ end }}
+finalize:
+  - from: /rootfs
+    to: /
+# Generating .kpxe via zbin:
```
Comment (Unix4ever) [inline ipxe/pkg.yaml:43]: is that some leftovers?

**Review by Unix4ever — APPROVED** [review]

File: ipxe/pkg.yaml (hunk 1 above)
Comment (smira) [reply ipxe/pkg.yaml:43]: no, that is doc on how to generate `.kpxe`.. 

so the thing is that we want to patch binary in place, but `.kpxe` is lzma-compressed; so we generate uncompressed version, a tool to generate compressed version, and we patch uncompressed version and compress it on the fly; thsee are the steps to compress the uncompressed version to build final `.kpxe`

**Pull request merged**